e2e/tests/truffle/contracts/TestLoomNativeApi.sol

```diff
@@ -0,0 +1,52 @@
+pragma solidity >=0.4.21;
+
+import "./LoomNativeApi.sol";
+
+library TestLoomNativeApi {
+    function TestMappedLoomAccount(string memory fromChain, string memory hash, uint8 v, bytes32 r, bytes32 s) public view returns (address) {
+        return LoomNativeApi.MappedAccount(fromChain, ecrecover(stripPrefix(hash), v, r, s));
```

Comment:
I suspect the reason you've had to add this `stripPrefix` thing here is that you're signing a hash of the message instead of the message itself:
```js
testHash = web3js.utils.sha3(msg);
sig = await wallet.signMessage(testHash)
```

If you did this instead I don't think you'll need to use `stripPrefix`:
```js
sig = await wallet.signMessage(msg);
testHash = web3js.utils.soliditySha3('\x19Ethereum Signed Message:\n32', msg);
```

Reply:
Actually we just spent a bunch of time doing something very similar today, here's how to get it working:
```js
sig = await wallet.signMessage(msg);
testHash = ethers.utils.hashMessage(msg);
```

Then you just pass the `testHash` to the Solidity contract and do:
```solidity
ecrecover(testHash, v, r, s)
```